crates/tauri-plugin-holochain-service/android/src/main/java/com/plugin/holochain_service/HolochainService.kt

```diff
@@ -131,6 +150,81 @@ class HolochainService : Service() {
             installedAppId: String
         ) {
             Log.d(TAG, "ensureAppWebsocket")
+            this.expectAuthorized()
+
+            serviceScope.launch(Dispatchers.IO) {
+                callback.ensureAppWebsocket(AppAuthFfiParcel(runtime?.ensureAppWebsocket(installedAppId)!!))
+            }
+        }
+
+        /// Sign a zome call
+        override fun signZomeCall(
+            callback: IHolochainServiceCallback,
+            req: ZomeCallUnsignedFfiParcel
+        ) {
+            Log.d(TAG, "signZomeCall")
+            this.expectAuthorized()
+
+            serviceScope.launch(Dispatchers.IO) {
+                callback.signZomeCall(ZomeCallFfiParcel(runtime!!.signZomeCall(req.inner)))
+            }
+        }
+
+        // We cannot call Binder.getCallingUid() within the onBind callback,
+        // so instead we check the authorization within each IPC call
+        private fun expectAuthorized() {
+            val clientUid = Binder.getCallingUid()
+            val clientPackageName = HolochainService@packageManager.getNameForUid(clientUid)
+                ?: throw UnauthorizedException("Unable to get name of package binding to AdminBinder uid=$clientUid")
+
+            Log.d(TAG, "AdminBinder expectAuthorized clientPackageName=$clientPackageName")
+
+            if (clientPackageName != HolochainService@packageName){
```

Comment:
Only the package that is running the HolochainService can call the admin binding.